src/components/settings-modal/settings-modal.scss

```diff
@@ -2,13 +2,17 @@
 @use '../../styles/mixins' as mixins;
 
 .kui-theme--light {
-  --color-modal-header-text: #{colors.$black-700};
+  --color-modal-header-text: #{colors.$black-0};
   --color-upgrade-reminder-text: #{colors.$blue-300};
+  --color-modal-table-text: rgb(0 0 0 / 85%);
+  --color-modal-table-description: rgb(0 0 0 / 70%);
 }
 
 .kui-theme--dark {
-  --color-modal-header-text: #{colors.$black-0};
+  --color-modal-header-text: #{colors.$black-300};
   --color-upgrade-reminder-text: #{colors.$yellow-300};
+  --color-modal-table-text: rgb(255 255 255 / 85%);
+  --color-modal-table-description: rgb(255 255 255 / 70%);
```

Comment:
You could check for rgb(...) in `_variables.scss` file instead of using rgb(...) color directly.